usecase/implementation/src/test/java/eu/openg/guessnumberapi/usecase/implementation/GuessNumberInteractorTest.java

```diff
@@ -0,0 +1,89 @@
+package eu.openg.guessnumberapi.usecase.implementation;
+
+import eu.openg.guessnumberapi.domain.GameEntity;
+import eu.openg.guessnumberapi.gateway.api.GameEntityRepository;
+import eu.openg.guessnumberapi.usecase.api.BoundaryGuessResponse;
+import eu.openg.guessnumberapi.usecase.api.BoundaryGuessResultStatus;
+import eu.openg.guessnumberapi.usecase.api.GuessNumberUseCase;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class GuessNumberInteractorTest {
+    @Mock
+    private GuessValidator gateway;
+    @Mock
+    private GameEntityRepository gameEntityRepository;
+
+    private GuessNumberUseCase guessNumberInteractor;
+
+    @BeforeEach
+    void setUp() {
+        guessNumberInteractor = new GuessNumberInteractor(gateway, gameEntityRepository);
+    }
+
+    @Test
+    void checkLowerGuessAndGuessResponse() {
+        checkGuessAndGuessResponse(false, 1, BoundaryGuessResultStatus.MORE);
+    }
+
+    @Test
+    void checkHigherGuessAndGuessResponse() {
+        checkGuessAndGuessResponse(true, 5, BoundaryGuessResultStatus.LESS);
+    }
+
+    private void checkGuessAndGuessResponse(boolean biggerThanGenerated, int guessedNumber,
+                                            BoundaryGuessResultStatus boundaryGuessResultStatus) {
+        when(gateway.isGuessBiggerThanGenerated(guessedNumber, 3)).thenReturn(biggerThanGenerated);
+        BoundaryGuessResponse response = checkGuessAndReturnGuessResponse(guessedNumber, false);
+        verify(gateway).isGuessBiggerThanGenerated(guessedNumber, 3);
+        assertEquals(response.getStatus(), boundaryGuessResultStatus);
+        assertNull(response.getNumberOfGuesses());
+    }
+
+    @Test
+    void checkCorrectGuessAndGuessResponse() {
+        BoundaryGuessResponse response = checkGuessAndReturnGuessResponse(3, true);
+        assertEquals(response.getNumberOfGuesses(), 1);
+        assertNull(response.getStatus());
```

Comment:
jeigu frontendas nori kažkokio keisto varianto, kad kai laimi tu jam nesakytum, kad laimi, tai tu tik reste taip padaryt, o pas save gali turėti normaliai. šiaip siūlyčiau susitarti, kad consistent būtų, visada siunti status, ir visada siunti number of guesses. būtų galima visą informaciją atvaizduoti, kad jau eina n-tasis spėjimas ar pnš